### DISCUSSION TITLE
How to solve multi objective optimization problem with high correlation

### DISCUSSION THREAD

**Discussion by JingDang99** [meta-pytorch/botorch#2278] [Q&A]
My optimization problem is two metrics, after correlation analysis, I found that these two metrics are strongly related, so I used MultiTaskGP.

However, I encountered RuntimeError: The size of tensor a (2) must match the size of tensor b (20) at non-singleton dimension 1
of tensor a (2) must match the size of tensor b (20) at nonsingleton dimension 1

My data structure is

print(train_X.shape)
print(train_Y.shape)

torch.Size([10, 3])
torch.Size([5, 2])

Translated with DeepL.com (free version)

**Comment by saitcakmak** [comment]
Hi @JingDang99. Seems like you're having difficulties with tensor shapes when using MultiTaskGP, though it is hard to say where these come from with the information you provided. I'd recommend referring to the docstring of MultiTaskGP for the expected tensor shapes: https://github.com/pytorch/botorch/blob/main/botorch/models/multitask.py#L155-L194
You can also check out this tutorial on how to utilize MultiTaskGP models: https://botorch.org/tutorials/composite_mtbo
